src/battle_terastal.c

```diff
@@ -47,10 +47,11 @@ bool32 CanTerastallize(u32 battler)
     u32 holdEffect = GetBattlerHoldEffect(battler, FALSE);
 
     // Check if Player has Tera Orb and has charge.
-    if (B_FLAG_TERA_ORB_CHARGED != 0
-        && (battler == B_POSITION_PLAYER_LEFT || (!(gBattleTypeFlags & BATTLE_TYPE_MULTI) && battler == B_POSITION_PLAYER_RIGHT))
+    if (!(B_FLAG_TERA_ORB_NO_COST != 0 && FlagGet(B_FLAG_TERA_ORB_NO_COST))
```

Comment:
Why is the check needed since the Tera Orb doesn't lose it's charge once `B_FLAG_TERA_ORB_NO_COST` is set?